pageserver/src/tenant/timeline.rs

```diff
@@ -1602,350 +1599,169 @@ impl Timeline {
-                local_layers
-            }
-        };
+        guard.initialize_local_layers(loaded_layers, disk_consistent_lsn + 1);
 
-        if has_local_layers {
-            // Are there local files that don't exist remotely? Schedule uploads for them.
-            // Local timeline metadata will get uploaded to remove along witht he layers.
-            for (layer_name, layer) in &local_only_layers {
-                // XXX solve this in the type system
-                let layer_path = layer
-                    .local_path()
-                    .expect("local_only_layers only contains local layers");
-                let layer_size = layer_path
-                    .metadata()
-                    .with_context(|| format!("failed to get file {layer_path:?} metadata"))?
-                    .len();
-                info!("scheduling {layer_path:?} for upload");
-                remote_client
-                    .schedule_layer_file_upload(layer_name, &LayerFileMetadata::new(layer_size))?;
+        if let Some(rtc) = self.remote_client.as_ref() {
+            if !needs_upload.is_empty() {
+                for (layer, m) in needs_upload {
+                    rtc.schedule_layer_file_upload(&layer.layer_desc().filename(), &m)?;
+                }
+                rtc.schedule_index_upload_for_file_changes()?;
             }
-            remote_client.schedule_index_upload_for_file_changes()?;
-        } else if index_part.is_none() {
-            // No data on the remote storage, no local layers, local metadata file.
-            //
-            // TODO https://github.com/neondatabase/neon/issues/3865
-            // Currently, console does not wait for the timeline data upload to the remote storage
-            // and considers the timeline created, expecting other pageserver nodes to work with it.
-            // Branch metadata upload could get interrupted (e.g pageserver got killed),
-            // hence any locally existing branch metadata with no remote counterpart should be uploaded,
-            // otherwise any other pageserver won't see the branch on `attach`.
-            //
-            // After the issue gets implemented, pageserver should rather remove the branch,
-            // since absence on S3 means we did not acknowledge the branch creation and console will have to retry,
-            // no need to keep the old files.
-            remote_client.schedule_index_upload_for_metadata_update(up_to_date_metadata)?;
-        } else {
-            // Local timeline has a metadata file, remote one too, both have no layers to sync.
         }
 
-        info!("Done");
+        info!(
+            "loaded layer map with {} layers at {}, total physical size: {}",
+            num_layers, disk_consistent_lsn, total_physical_size
+        );
+        self.metrics
+            .resident_physical_size_gauge
+            .set(total_physical_size);
```

Comment:
Don't we need to set the remote_pysical_size_gauge as well somewhere? Would feel natural to do it here.

Reply:
I think that is managed by remote_timeline_client whenever index_part is uploaded..